scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -904,38 +870,30 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   }
   def atPos[T <: Tree](start: Pos, end: Pos)(body: => T): T = {
     val startTokenPos = start.startTokenPos
-    val result = body
-    var endTokenPos = end.endTokenPos
-    if (endTokenPos < startTokenPos) endTokenPos = startTokenPos - 1
-    val pos = trimmedPos(startTokenPos, endTokenPos)
-    result.withOrigin(Origin.Parsed(input, dialect, pos)).asInstanceOf[T]
+    atPosWithBody(startTokenPos, body, end)
   }
 
-  def trimmedPos(startTokenPos: Int, endTokenPos: Int): TokenStreamPosition = {
-    def skipTrivia(range: Range): Int = {
-      range
-        .dropWhile(i => scannerTokens(i).is[Trivia])
-        .headOption
-        .getOrElse(range.start)
-    }
+  def atPosWithBody[T <: Tree](startTokenPos: Int, body: T, endPos: Pos): T = {
+    var endTokenPos = endPos.endTokenPos
+    if (endTokenPos < startTokenPos) endTokenPos = startTokenPos - 1
+
+    def skipBy(range: Range, f: Token => Boolean): Int =
+      range.dropWhile(i => f(scannerTokens(i))).headOption.getOrElse(range.start)
+    @inline def skipTrivia(range: Range): Int = skipBy(range, _.is[Trivia])
```

Comment:
`skipBy` is called only from `skipTrivia`. Maybe merge them into a single func?

Reply:
i intentionally did this. in the next pr i will define skipping by whitespace only, keeping comments.